src/main/java/org/elasticsearch/script/expression/FieldDataValueSource.java

```diff
@@ -49,13 +48,24 @@ public FunctionValues getValues(Map context, LeafReaderContext leaf) throws IOEx
     }
 
     @Override
-    public boolean equals(Object other) {
-        return fieldData.equals(other);
+    public boolean equals(Object o) {
```

Comment:
Thanks for regenerating, I was only referring to the null checks that were in the return statement before!